src/txdb.cpp

```diff
@@ -66,19 +66,18 @@ CCoinsViewDB::CCoinsViewDB(size_t nCacheSize, bool fMemory, bool fWipe)
 
 bool CCoinsViewDB::GetCoin(const COutPoint &outpoint, Coin &coin) const
 {
-    LOCK(cs_utxo);
+    READLOCK(cs_utxo);
     return db.Read(CoinEntry(&outpoint), coin);
 }
 
 bool CCoinsViewDB::HaveCoin(const COutPoint &outpoint) const
 {
-    LOCK(cs_utxo);
+    READLOCK(cs_utxo);
     return db.Exists(CoinEntry(&outpoint));
 }
 
 uint256 CCoinsViewDB::GetBestBlock() const
 {
-    LOCK(cs_utxo);
```

Comment:
READLOCK here? Maybe is taken somewhere else, if this is case I wonder if we could add a comment about it.

Reply:
i made the same comment, it's taken somewhere else...

Reply:
The code is correct because the lock is taken in the derived (ViewCache) class.  But you guys are right that we have a code clarity problem.  And if the ViewDB was ever used directly then the lock would not be taken.  

I think I need locked and lockless versions in the base class to resolve this.